### PR TITLE
Update Rust client guide

### PR DESCRIPTION
**Pull request by noahcrowley**
Updated Rust client guide.

I'm not sure that I've dealt with error handling in the examples in the most Rust-like way. From what I could find, `unwrap` is often used in examples to skip error handling, so that's what I went with, but would appreciate a sanity check from someone who writes Rust.

Also, I ended up using 3.0.0-beta.3 instead of 2.1.3, which has dependencies that don't work with current versions of OpenSSL.

### CODE REVIEW

**Comment by bnlb** [comment]
Added @anowell as a reviewer as he knows Rust better than I do.

File: _pages/clients/rust.md [hunk 1]
```diff
@@ -63,200 +61,200 @@ If you are running [Algorithmia Enterprise](/enterprise), you can specify the AP
...
+
+If the file was text (an image, etc.), you can use `read_to_string` on the response. The [API Specification](/developers/api/#get-a-file-or-directory) has more details on how to get files from a data collection using the Data API.
+
+## Additional Functionality
+
+In addition to the functionality covered in this guide, the Rust Client Library provides a complete interface to the Algorithmia platform, including [managing algorithms](/developers/algorithm-development/algorithm-management), administering [organizations](/developers/platform/organizations), and working with [source control](/developers/algorithm-development/source-code-management). You can also visit the [API Docs](/developers/api) to view the complete API specification.
+
+## Next Steps
+
+If you're a data scientist or developer who will be building and deploying new algorithms, you can move on to the [Algorithm Development > Getting Started](/developers/algorithm-development/your-first-algo/) guide.
```
Comment (bnlb) [inline _pages/clients/rust.md]:
```suggestion
If you're a data scientist or developer who will be building and deploying new algorithms, you can move on to the [Algorithm Development > Getting Started](/developers/algorithm-development/your-first-algo) guide.
```

**Review by bnlb — DISMISSED** [review]
lgtm! Will defer to Anthony on the Rust code snippets

File: _pages/clients/rust.md (hunk 1 above)
Comment (noahcrowley) [reply _pages/clients/rust.md]: Gotta remember the trailing slashes...

Comment (bnlb) [reply _pages/clients/rust.md]: lol, we had a huge push a year ago to get that all sorted out. Was several sprints worth of work to optimize across several different apps. It's really not a big deal if one slips through, they're just burned into my brain and they stick out to me now.

**Review by bnlb — DISMISSED** [review]

File: _pages/clients/rust.md [hunk 2]
```diff
@@ -63,200 +61,200 @@ If you are running [Algorithmia Enterprise](/enterprise), you can specify the AP
...
+The documentation for "Face Detection" says that it takes a URL or a Data URI of the image to be processed, and a Data URI where the algorithm can store the result. We'll create a directory to host the input image and set its [permissions](/developers/api/#update-collection-acl) to make it publicly accessible by creating a DataAcl with `DataAcl::from(ReadAcl::Public)` and passing it to the `create` method.
 
 {% highlight rust %}
-// Download contents of file as a string
-if client.file(text_file).exists().unwrap() {
-	let mut text_reader = client.file(text_file).get().unwrap();
-	let mut jack_london_text = String::new();
-	let _ = text_reader.read_to_string(&mut jack_london_text);
+let img_directory = client.dir("data://YOUR_USERNAME/img_directory");
+if img_directory.exists().unwrap() == false {
```
Comment (anowell) [inline _pages/clients/rust.md]: nit: testing for `== false` followed by an `else` clause is less natural to read than:

```
if img_directory.exists()? {
   println!("img_directory exists");
} else { 
    img_directory.create(DataAcl::from(ReadAcl::Public))?;
}
```

(no strong opinion here on `?` vs `unwrap()` in example code)

Comment (noahcrowley) [reply _pages/clients/rust.md]: Looks good, updated in two places, thanks!

**Review by bnlb — APPROVED** [review]
👍

**Pull request merged**